### DISCUSSION TITLE
Search Scopes & Extending the Search View

### DISCUSSION THREAD

**Discussion by prateek** [microsoft/vscode-discussions#531] [Extension Development QnA] [answered]
Hey folks, I've been VSCode since 2016 - primarily for Go development. One source of friction I've noticed is I'm often changing the scope of which files I'm searching. 

I added my thoughts to the relevant Feature Request PR: https://github.com/microsoft/vscode/issues/41860#issuecomment-1474925916. Orthogonal to that discussion, I wanted to explore if it was possible for me to implement the feature myself. I haven't made any extensions so I don't know what's possible, hence this thread. Please let me know if my question doesn't belong here.

My questions/requests:
- Can I extend/manipulate existing views in VSCode extensions? E.g. I want to change visual attributes provided in the Search view: adding a new "Scope" input box
- Any resources you can point me at to go learn/figure this out myself? I'll accept if the answer is "Source Code" but I'm hoping for something better.

Thanks!

**Comment by digitarald** [comment, marked as answer]
Extending the search view itself is not possible.

Best way forward would be to contribute this feature in https://github.com/microsoft/vscode/issues/41860#issuecomment-1474925916 , but it probably hasn't been picked up yet due to the hidden complexity as this would need to be applied across different parts of VS Code (explorer, quick picks, search, etc).
